cloud/docker/docker_login.py

```diff
@@ -0,0 +1,243 @@
+
+- name: login to DockerHub remote registry using your account
+  docker_login:
+    username: docker
+    password: rekcod
+    email: docker@docker.io
+
+- name: login to private Docker remote registry and force reauthentification
+  docker_login:
+    registry: https://your.private.registry.io/v1/
+    username: yourself
+    password: secrets3
+    reauth: yes
+
+- name: login to DockerHub remote registry using a custom dockercfg file location
+  docker_login:
+    username: docker
+    password: rekcod
+    email: docker@docker.io
+    dockercfg_path: /tmp/.mydockercfg
+
+'''
+
+try:
+    import os.path
+    import sys
+    import json
+    import base64
+    import docker.client
+    from requests.exceptions import *
+    from urlparse import urlparse
+except ImportError, e:
+    print "failed=True msg='failed to import python module: %s'" % e
+    sys.exit(1)
+
+try:
+    from docker.errors import APIError as DockerAPIError
+except ImportError:
+    from docker.client import APIError as DockerAPIError
+
+class DockerLoginManager:
+
+    def __init__(self, module):
+
+        self.module = module
+        self.registry = self.module.params.get('registry')
+        self.username = self.module.params.get('username')
+        self.password = self.module.params.get('password')
+        self.email = self.module.params.get('email')
+        self.reauth = self.module.params.get('reauth')
+        self.dockercfg_path = os.path.expanduser(self.module.params.get('dockercfg_path'))
+
+        docker_url = urlparse(module.params.get('docker_url'))
+        self.client = docker.Client(base_url=docker_url.geturl(), timeout=module.params.get('timeout'))
+
+        self.changed = False
+        self.response = False
+        self.log = list()
+
+    def login(self):
+
+        if self.reauth:
+            self.log.append("Enforcing reauthentification")
+
+        # Connect to registry and login if not already logged in or reauth is enforced.
+        try:
+            self.response = self.client.login(
+                self.username,
+                password=self.password,
+                email=self.email,
+                registry=self.registry,
+                reauth=self.reauth,
+                dockercfg_path=self.dockercfg_path
+            )
+        except Exception as e:
+            self.module.fail_json(msg="failed to login to the remote registry", error=repr(e))
+
+        # Get status from registry response.
+        if self.response.has_key("Status"):
+            self.log.append(self.response["Status"])
+            if self.response["Status"] == "Login Succeeded":
+                self.changed = True
+        else:
+            self.log.append("Already Authentificated")
+
+        # Update the dockercfg if changed but not failed.
+        if self.has_changed():
+            self.update_dockercfg()
```

Comment:
If I understand this right, It would be easy to implement check mode here if you add 

~~~python
if self.has_changed() and not self.module.check_mode: 
     self.update_dockercfg()
~~~

and don't forget to add `supports_check_mode=True` just like this exmaple. https://github.com/ansible/ansible-modules-extras/blob/devel/cloud/cloudstack/cs_affinitygroup.py#L206